--- youtube-dl/youtube_dl/YoutubeDL.py.orig	2013-10-23 07:16:26.000000000 +0900
+++ youtube-dl/youtube_dl/YoutubeDL.py	2013-10-23 13:10:41.699818669 +0900
@@ -9,6 +9,7 @@
 import re
 import shutil
 import socket
+import string
 import sys
 import time
 import traceback
@@ -261,6 +262,22 @@
         """Increment the ordinal that assigns a number to each file."""
         self._num_downloads += 1
 
+    def unsafe2safe(self, string):
+        convert_tuples = [
+            (u'\u00a6', u'\u007c'), # broken bar  -> vertical bar
+            (u'\u2014', u'\u2015'), # em dash     -> horizontal bar
+            (u'\u2225', u'\u2016'), # parallel to -> double vertical line
+            (u'\uff0d', u'\u2212'), # fullwidth hyphen-minus -> minus sign
+            (u'\uff5e', u'\u301c'), # fullwidth tilde        -> wave dash
+            (u'\uffe0', u'\u00a2'), # fullwidth cent sign    -> cent sign
+            (u'\uffe1', u'\u00a3'), # fullwidth pound sign   -> pound sign
+            (u'\uffe2', u'\u00ac')  # fullwidth not sign     -> not sign
+        ]
+
+        for unsafe, safe in convert_tuples:
+            string = string.replace(unsafe, safe)
+        return string
+
     def prepare_filename(self, info_dict):
         """Generate the output filename."""
         try:
@@ -283,7 +300,7 @@
                                  for k, v in template_dict.items())
 
             tmpl = os.path.expanduser(self.params['outtmpl'])
-            filename = tmpl % template_dict
+            filename = self.unsafe2safe(tmpl % template_dict)
             return filename
         except KeyError as err:
             self.report_error(u'Erroneous output template')
