*** libreoffice-6.2.4.2/sw/qa/extras/odfexport/odfexport.cxx~	2019-05-18 00:35:40.000000000 +0900
--- libreoffice-6.2.4.2/sw/qa/extras/odfexport/odfexport.cxx	2019-05-26 11:15:11.033077324 +0900
***************
*** 1950,1956 ****
      // (used from LibreOffice 6.1, and proposed for next ODF)
      const char* aFieldTexts[] = { "A 2", "Az Isten", "Az 50-esek",
          "A 2018-asok", "Az egyebek", "A fejezetek",
!         u8"Az „Őseinket...”", "a 2",
          "Az v", "az 1", "Az e", "az 1",
          "Az (5)", "az 1", "A 2", "az 1" };
      uno::Reference<text::XTextFieldsSupplier> xTextFieldsSupplier(mxComponent, uno::UNO_QUERY);
--- 1950,1956 ----
      // (used from LibreOffice 6.1, and proposed for next ODF)
      const char* aFieldTexts[] = { "A 2", "Az Isten", "Az 50-esek",
          "A 2018-asok", "Az egyebek", "A fejezetek",
!         "Az „Őseinket...”", "a 2",
          "Az v", "az 1", "Az e", "az 1",
          "Az (5)", "az 1", "A 2", "az 1" };
      uno::Reference<text::XTextFieldsSupplier> xTextFieldsSupplier(mxComponent, uno::UNO_QUERY);
