diff -Nrc libreoffice-6.0.5.2/pyuno/source/module/pyuno_impl.hxx build/pyuno/source/module/pyuno_impl.hxx
*** libreoffice-6.0.5.2/pyuno/source/module/pyuno_impl.hxx	2018-06-20 07:27:30.000000000 +0900
--- build/pyuno/source/module/pyuno_impl.hxx	2018-07-22 06:55:56.941974177 +0900
***************
*** 80,86 ****
      return PyUnicode_FromString(string);
  }
  
! inline char * PyStr_AsString(PyObject *object)
  {
      return PyUnicode_AsUTF8(object);
  }
--- 80,86 ----
      return PyUnicode_FromString(string);
  }
  
! inline const char * PyStr_AsString(PyObject *object)
  {
      return PyUnicode_AsUTF8(object);
  }
diff -Nrc libreoffice-6.0.5.2/pyuno/source/module/pyuno_type.cxx build/pyuno/source/module/pyuno_type.cxx
*** libreoffice-6.0.5.2/pyuno/source/module/pyuno_type.cxx	2018-06-20 07:27:30.000000000 +0900
--- build/pyuno/source/module/pyuno_type.cxx	2018-07-22 06:58:58.329968426 +0900
***************
*** 157,163 ****
      }
  
      OUString strTypeName( OUString::createFromAscii( PyStr_AsString( typeName.get() ) ) );
!     char *stringValue = PyStr_AsString( value.get() );
  
      TypeDescription desc( strTypeName );
      if( !desc.is() )
--- 157,163 ----
      }
  
      OUString strTypeName( OUString::createFromAscii( PyStr_AsString( typeName.get() ) ) );
!     const char *stringValue = PyStr_AsString( value.get() );
  
      TypeDescription desc( strTypeName );
      if( !desc.is() )
diff -Nrc libreoffice-6.0.5.2/pyuno/source/module/pyuno_util.cxx build/pyuno/source/module/pyuno_util.cxx
*** libreoffice-6.0.5.2/pyuno/source/module/pyuno_util.cxx	2018-06-20 07:27:30.000000000 +0900
--- build/pyuno/source/module/pyuno_util.cxx	2018-07-22 06:58:28.470969373 +0900
***************
*** 69,75 ****
  #else
  #if PY_MAJOR_VERSION >= 3
      Py_ssize_t size(0);
!     char *pUtf8(PyUnicode_AsUTF8AndSize(pystr, &size));
      ret = OUString(pUtf8, size, RTL_TEXTENCODING_UTF8);
  #else
      PyObject* pUtf8 = PyUnicode_AsUTF8String(pystr);
--- 69,75 ----
  #else
  #if PY_MAJOR_VERSION >= 3
      Py_ssize_t size(0);
!     const char *pUtf8(PyUnicode_AsUTF8AndSize(pystr, &size));
      ret = OUString(pUtf8, size, RTL_TEXTENCODING_UTF8);
  #else
      PyObject* pUtf8 = PyUnicode_AsUTF8String(pystr);
