*** lame-3.99.5/configure	2012-02-29 03:54:37.000000000 +0900
--- build32/configure	2017-01-28 07:01:41.252747628 +0900
***************
*** 11913,11918 ****
--- 11913,11927 ----
  
  fi
  
+ case $host_cpu in
+ i686*)
+  check_xmmintrin=
+  ;;
+ *)
+  check_xmmintrin="xmmintrin.h"
+  ;;
+ esac
+ 
  for ac_header in \
  		 errno.h \
  		 fcntl.h \
***************
*** 11922,11928 ****
  		 sys/soundcard.h \
  		 sys/time.h \
  		 unistd.h \
! 		 xmmintrin.h \
  		 linux/soundcard.h
  do :
    as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
--- 11931,11937 ----
  		 sys/soundcard.h \
  		 sys/time.h \
  		 unistd.h \
! 		 $check_xmmintrin \
  		 linux/soundcard.h
  do :
    as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
