*** firefox-78.4.0/js/src/wasm/WasmSignalHandlers.cpp~	2020-10-14 18:34:46.000000000 +0900
--- firefox-78.4.0/js/src/wasm/WasmSignalHandlers.cpp	2020-11-03 15:09:37.837020741 +0900
***************
*** 249,255 ****
  #endif
  
  #ifdef WASM_EMULATE_ARM_UNALIGNED_FP_ACCESS
! #  include <sys/user.h>
  #endif
  
  #if defined(ANDROID)
--- 249,272 ----
  #endif
  
  #ifdef WASM_EMULATE_ARM_UNALIGNED_FP_ACCESS
! /*
!  * User specific VFP registers. If only VFPv2 is present, registers 16 to 31
!  * are ignored by the ptrace system call and the signal handler.
!  */
! struct user_vfp {
!  unsigned long long fpregs[32];
!  unsigned long fpscr;
! };
! 
! /*
!  * VFP exception registers exposed to user space during signal delivery.
!  * Fields not relavant to the current VFP architecture are ignored.
!  */
! struct user_vfp_exc {
!  unsigned long fpexc;
!  unsigned long fpinst;
!  unsigned long fpinst2;
! };
  #endif
  
  #if defined(ANDROID)
