*** ftgl-2.1.3~rc5/demo/Makefile.am	2008-06-02 10:10:10.000000000 +0900
--- build/demo/Makefile.am	2017-01-13 13:40:01.916659773 +0900
***************
*** 9,22 ****
      simple.cpp \
      $(NULL)
  simple_CXXFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
! simple_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS)
  simple_LDADD = ../src/libftgl.la
  
  c_demo_SOURCES = \
      c-demo.c \
      $(NULL)
  c_demo_CFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
! c_demo_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS)
  c_demo_LDADD = ../src/libftgl.la
  
  FTGLDemo_SOURCES = \
--- 9,22 ----
      simple.cpp \
      $(NULL)
  simple_CXXFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
! simple_LDFLAGS = $(FT2_LIBS) $(GL_LIBS) $(GLUT_LIBS) -lm
  simple_LDADD = ../src/libftgl.la
  
  c_demo_SOURCES = \
      c-demo.c \
      $(NULL)
  c_demo_CFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
! c_demo_LDFLAGS = $(FT2_LIBS) $(GL_LIBS) $(GLUT_LIBS) -lm
  c_demo_LDADD = ../src/libftgl.la
  
  FTGLDemo_SOURCES = \
***************
*** 27,33 ****
  	trackball.h \
  	$(NULL)
  FTGLDemo_CXXFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
! FTGLDemo_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS)
  FTGLDemo_LDADD = ../src/libftgl.la
  
  FTGLMFontDemo_SOURCES = \
--- 27,33 ----
  	trackball.h \
  	$(NULL)
  FTGLDemo_CXXFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
! FTGLDemo_LDFLAGS = $(FT2_LIBS) $(GL_LIBS) $(GLUT_LIBS) -lm
  FTGLDemo_LDADD = ../src/libftgl.la
  
  FTGLMFontDemo_SOURCES = \
***************
*** 38,44 ****
  	trackball.h \
  	$(NULL)
  FTGLMFontDemo_CXXFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
! FTGLMFontDemo_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS)
  FTGLMFontDemo_LDADD = ../src/libftgl.la
  
  NULL =
--- 38,44 ----
  	trackball.h \
  	$(NULL)
  FTGLMFontDemo_CXXFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
! FTGLMFontDemo_LDFLAGS = $(FT2_LIBS) $(GL_LIBS) $(GLUT_LIBS) -lm
  FTGLMFontDemo_LDADD = ../src/libftgl.la
  
  NULL =
*** ftgl-2.1.3~rc5/demo/Makefile.in	2008-06-12 23:33:01.000000000 +0900
--- build/demo/Makefile.in	2017-01-13 13:43:25.996659597 +0900
***************
*** 242,255 ****
      $(NULL)
  
  simple_CXXFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
! simple_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS)
  simple_LDADD = ../src/libftgl.la
  c_demo_SOURCES = \
      c-demo.c \
      $(NULL)
  
  c_demo_CFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
! c_demo_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS)
  c_demo_LDADD = ../src/libftgl.la
  FTGLDemo_SOURCES = \
  	FTGLDemo.cpp \
--- 242,255 ----
      $(NULL)
  
  simple_CXXFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
! simple_LDFLAGS = $(FT2_LIBS) $(GL_LIBS) $(GLUT_LIBS) -lm
  simple_LDADD = ../src/libftgl.la
  c_demo_SOURCES = \
      c-demo.c \
      $(NULL)
  
  c_demo_CFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
! c_demo_LDFLAGS = $(FT2_LIBS) $(GL_LIBS) $(GLUT_LIBS) -lm
  c_demo_LDADD = ../src/libftgl.la
  FTGLDemo_SOURCES = \
  	FTGLDemo.cpp \
***************
*** 260,266 ****
  	$(NULL)
  
  FTGLDemo_CXXFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
! FTGLDemo_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS)
  FTGLDemo_LDADD = ../src/libftgl.la
  FTGLMFontDemo_SOURCES = \
  	FTGLMFontDemo.cpp \
--- 260,266 ----
  	$(NULL)
  
  FTGLDemo_CXXFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
! FTGLDemo_LDFLAGS = $(FT2_LIBS) $(GL_LIBS) $(GLUT_LIBS) -lm
  FTGLDemo_LDADD = ../src/libftgl.la
  FTGLMFontDemo_SOURCES = \
  	FTGLMFontDemo.cpp \
***************
*** 271,277 ****
  	$(NULL)
  
  FTGLMFontDemo_CXXFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
! FTGLMFontDemo_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS)
  FTGLMFontDemo_LDADD = ../src/libftgl.la
  NULL = 
  all: all-am
--- 271,277 ----
  	$(NULL)
  
  FTGLMFontDemo_CXXFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
! FTGLMFontDemo_LDFLAGS = $(FT2_LIBS) $(GL_LIBS) $(GLUT_LIBS) -lm
  FTGLMFontDemo_LDADD = ../src/libftgl.la
  NULL = 
  all: all-am
