diff -Nrc konqueror-18.04.0.orig/plugins/validators/tidy_validator.cpp konqueror-18.04.0/plugins/validators/tidy_validator.cpp
*** konqueror-18.04.0.orig/plugins/validators/tidy_validator.cpp	2018-04-03 14:12:48.000000000 +0900
--- konqueror-18.04.0/plugins/validators/tidy_validator.cpp	2018-07-20 07:33:10.948878398 +0900
***************
*** 25,31 ****
  
  #include <kdebug.h>
  
! #include <buffio.h>
  #include <tidy.h>
  
  #include <config-konq-validator.h>
--- 25,31 ----
  
  #include <kdebug.h>
  
! #include <tidybuffio.h>
  #include <tidy.h>
  
  #include <config-konq-validator.h>
