*** TiMidity++-2.14.0/timidity/mfi.c	2004-02-18 01:02:20.000000000 +0900
--- build/timidity/mfi.c	2017-01-17 15:30:35.763388854 +0900
***************
*** 344,350 ****
  #define SEND_LASTNOTEINFO(lni, ch)				if (LASTNOTEINFO_HAS_DATA((lni)[ch])) SendLastNoteInfo(lni, ch);
  #define SEND_AND_CLEAR_LASTNOTEINFO(lni, ch)	if (LASTNOTEINFO_HAS_DATA((lni)[ch])) { SendLastNoteInfo(lni, ch); (lni)[ch].on = NO_LAST_NOTE_INFO; }
  
! inline void StoreLastNoteInfo(LastNoteInfo *info, int channel, int time, int duration, int note, int velocity)
  {
  	info[channel].on = time;
  	info[channel].off = time + duration;
--- 344,350 ----
  #define SEND_LASTNOTEINFO(lni, ch)				if (LASTNOTEINFO_HAS_DATA((lni)[ch])) SendLastNoteInfo(lni, ch);
  #define SEND_AND_CLEAR_LASTNOTEINFO(lni, ch)	if (LASTNOTEINFO_HAS_DATA((lni)[ch])) { SendLastNoteInfo(lni, ch); (lni)[ch].on = NO_LAST_NOTE_INFO; }
  
! static inline void StoreLastNoteInfo(LastNoteInfo *info, int channel, int time, int duration, int note, int velocity)
  {
  	info[channel].on = time;
  	info[channel].off = time + duration;
***************
*** 352,358 ****
  	info[channel].velocity = velocity;
  }
  
! inline void SendLastNoteInfo(const LastNoteInfo *info, int channel)
  {
  	NOTE_BUF_EV_DEBUGSTR(channel, info[channel].on, note_name[info[channel].note % 12], info[channel].note / 12, info[channel].velocity, info[channel].off);
  	MIDIEVENT(info[channel].on, ME_NOTEON, channel, info[channel].note, info[channel].velocity);
--- 352,358 ----
  	info[channel].velocity = velocity;
  }
  
! static inline void SendLastNoteInfo(const LastNoteInfo *info, int channel)
  {
  	NOTE_BUF_EV_DEBUGSTR(channel, info[channel].on, note_name[info[channel].note % 12], info[channel].note / 12, info[channel].velocity, info[channel].off);
  	MIDIEVENT(info[channel].on, ME_NOTEON, channel, info[channel].note, info[channel].velocity);
