*** libcap-2.25/libcap/Makefile	2016-01-31 09:01:41.000000000 +0900
--- build/libcap/Makefile	2017-02-14 04:21:55.763233818 +0900
***************
*** 22,28 ****
  
  ifeq ($(BUILD_GPERF),yes)
  USE_GPERF_OUTPUT = $(GPERF_OUTPUT)
! INCLUDE_GPERF_OUTPUT = -include $(GPERF_OUTPUT)
  endif
  
  libcap.pc: libcap.pc.in
--- 22,28 ----
  
  ifeq ($(BUILD_GPERF),yes)
  USE_GPERF_OUTPUT = $(GPERF_OUTPUT)
! INCLUDE_GPERF_OUTPUT = -include stddef.h -include $(GPERF_OUTPUT)
  endif
  
  libcap.pc: libcap.pc.in
***************
*** 41,47 ****
  	./_makenames > cap_names.h
  
  $(GPERF_OUTPUT): cap_names.list.h
! 	perl -e 'print "struct __cap_token_s { const char *name; int index; };\n%{\nconst struct __cap_token_s *__cap_lookup_name(const char *, unsigned int);\n%}\n%%\n"; while ($$l = <>) { $$l =~ s/[\{\"]//g; $$l =~ s/\}.*// ; print $$l; }' < $< | gperf --ignore-case --language=ANSI-C --readonly --null-strings --global-table --hash-function-name=__cap_hash_name --lookup-function-name="__cap_lookup_name" -c -t -m20 $(INDENT) > $@
  
  cap_names.list.h: Makefile $(KERNEL_HEADERS)/linux/capability.h
  	@echo "=> making $@ from $(KERNEL_HEADERS)/linux/capability.h"
--- 41,47 ----
  	./_makenames > cap_names.h
  
  $(GPERF_OUTPUT): cap_names.list.h
! 	perl -e 'print "struct __cap_token_s { const char *name; int index; };\n%{\nconst struct __cap_token_s *__cap_lookup_name(const char *, size_t);\n%}\n%%\n"; while ($$l = <>) { $$l =~ s/[\{\"]//g; $$l =~ s/\}.*// ; print $$l; }' < $< | gperf --ignore-case --language=ANSI-C --readonly --null-strings --global-table --hash-function-name=__cap_hash_name --lookup-function-name="__cap_lookup_name" -c -t -m20 $(INDENT) > $@
  
  cap_names.list.h: Makefile $(KERNEL_HEADERS)/linux/capability.h
  	@echo "=> making $@ from $(KERNEL_HEADERS)/linux/capability.h"
