*** freetype-2.7.1/modules.cfg	2016-12-21 03:49:00.000000000 +0900
--- build/modules.cfg	2017-05-01 00:46:19.076706387 +0900
***************
*** 120,126 ****
  # TrueType GX/AAT table validation.  Needs ftgxval.c below.
  #
  # No FT_CONFIG_OPTION_PIC support.
! # AUX_MODULES += gxvalid
  
  # Support for streams compressed with gzip (files with suffix .gz).
  #
--- 120,126 ----
  # TrueType GX/AAT table validation.  Needs ftgxval.c below.
  #
  # No FT_CONFIG_OPTION_PIC support.
! AUX_MODULES += gxvalid
  
  # Support for streams compressed with gzip (files with suffix .gz).
  #
***************
*** 143,149 ****
  # OpenType table validation.  Needs ftotval.c below.
  #
  # No FT_CONFIG_OPTION_PIC support.
! # AUX_MODULES += otvalid
  
  # Auxiliary PostScript driver component to share common code.
  #
--- 143,149 ----
  # OpenType table validation.  Needs ftotval.c below.
  #
  # No FT_CONFIG_OPTION_PIC support.
! AUX_MODULES += otvalid
  
  # Auxiliary PostScript driver component to share common code.
  #
