*** freetype-2.7.1/include/freetype/config/ftoption.h	2016-12-11 15:53:49.000000000 +0900
--- build/include/freetype/config/ftoption.h	2017-05-01 00:49:53.610052148 +0900
***************
*** 122,128 ****
    /* This is done to allow FreeType clients to run unmodified, forcing     */
    /* them to display normal gray-level anti-aliased glyphs.                */
    /*                                                                       */
! /* #define FT_CONFIG_OPTION_SUBPIXEL_RENDERING */
  
  
    /*************************************************************************/
--- 122,128 ----
    /* This is done to allow FreeType clients to run unmodified, forcing     */
    /* them to display normal gray-level anti-aliased glyphs.                */
    /*                                                                       */
! #define FT_CONFIG_OPTION_SUBPIXEL_RENDERING
  
  
    /*************************************************************************/
