*** xv-3.10a.orig/xv.c	Fri Jan 20 03:08:43 1995
--- xv-3.10a/xv.c	Sat Dec  2 18:58:22 2000
***************
*** 360,366 ****
  	}
        }
      }
!     
      if (best>=0 && best<numvis) useOtherVisual(vinfo, best);
      if (vinfo) XFree((char *) vinfo);
    }
--- 360,370 ----
  	}
        }
      }
!     if( best>=0 && best<numvis ){
! 	if( vinfo[best].depth==DefaultDepth(theDisp,DefaultScreen(theDisp)) ){
! 	    best=-1 ;
! 	}
!     }
      if (best>=0 && best<numvis) useOtherVisual(vinfo, best);
      if (vinfo) XFree((char *) vinfo);
    }
