*** xv-3.10a.orig/xvgrab.c	Fri Dec 23 07:34:47 1994
--- xv-3.10a/xvgrab.c	Sat Dec  2 19:00:16 2000
***************
*** 568,575 ****
      if (isLsbMachine) fprintf(stderr,"This looks like an lsbfirst machine\n");
                   else fprintf(stderr,"This looks like an msbfirst machine\n");
    }
! 
! 
    if (image->bitmap_unit != 8 && image->bitmap_unit != 16 &&
        image->bitmap_unit != 32) {
      sprintf(errstr, "%s\nReturned image bitmap_unit (%d) non-standard.",
--- 568,575 ----
      if (isLsbMachine) fprintf(stderr,"This looks like an lsbfirst machine\n");
                   else fprintf(stderr,"This looks like an msbfirst machine\n");
    }
!   if( 16 < image->depth ) image->bitmap_unit=32 ;
!   if( image->depth ==16 ) image->bitmap_unit=16 ;
    if (image->bitmap_unit != 8 && image->bitmap_unit != 16 &&
        image->bitmap_unit != 32) {
      sprintf(errstr, "%s\nReturned image bitmap_unit (%d) non-standard.",
