*** qpopper4.1.0/popper/popper.h	2011-05-31 04:13:40.000000000 +0900
--- build/popper/popper.h	2017-01-14 12:22:35.526646333 +0900
***************
*** 310,327 ****
  
  extern int              errno;
  
- #if !( defined(BSD) && (BSD >= 199306) ) && !defined(__USE_BSD)
-    extern int              sys_nerr;
- #  ifndef FREEBSD
-      extern char         *   sys_errlist[];
- #    ifndef SYS_SIGLIST_DECLARED
- #      ifndef __linux__
-          extern char         *   sys_siglist[];
- #      endif /* not linux */
- #    endif /* not SYS_SIGLIST_DECLARED */
- #  endif /* not FREEBSD */
- #endif /* not BSD >= 199306 and not __USE_BSD */
- 
  extern int               pop_timeout;
  
  extern volatile BOOL     hangup;
--- 310,315 ----
