*** audex-0.79/utils/encoderassistant.cpp.orig	Wed Nov  2 18:23:48 2016
--- audex-0.79/utils/encoderassistant.cpp	Wed Nov  2 18:26:35 2016
***************
*** 256,262 ****
          if (embed_cover) {
            cmd += QString::fromUtf8(" ${"VAR_COVER_FILE" pre=\"--add-image=\" post=\":FRONT_COVER\"}");
          }
! 	cmd += QString::fromUtf8(" --set-encoding=latin1 $"VAR_OUTPUT_FILE" && eyeD3 --to-v2.3 $"VAR_OUTPUT_FILE);
  
        } else {
  
--- 256,263 ----
          if (embed_cover) {
            cmd += QString::fromUtf8(" ${"VAR_COVER_FILE" pre=\"--add-image=\" post=\":FRONT_COVER\"}");
          }
! 	// cmd += QString::fromUtf8(" --set-encoding=latin1 $"VAR_OUTPUT_FILE" && eyeD3 --to-v2.3 $"VAR_OUTPUT_FILE);
! 	cmd += QString::fromUtf8(" --set-encoding=utf8 $"VAR_OUTPUT_FILE" && eyeD3 --to-v2.3 $"VAR_OUTPUT_FILE);
  
        } else {
  
