diff -Nrc efibootmgr-devel-efibootmgr-0.12/src/efibootmgr/efibootmgr.c efibootmgr-devel-efibootmgr-0.12/src/efibootmgr/efibootmgr.c
*** efibootmgr-devel-efibootmgr-0.12/src/efibootmgr/efibootmgr.c	Thu May 28 23:23:42 2015
--- efibootmgr-devel-efibootmgr-0.12/src/efibootmgr/efibootmgr.c	Fri Nov 25 10:51:54 2016
***************
*** 58,63 ****
--- 58,64 ----
  #define EFIBOOTMGR_VERSION "unknown (fix Makefile!)"
  #endif
  
+ efi_guid_t const efi_guid_zero = {0};
  
  typedef struct _var_entry {
  	char		*name;
