diff -Nrc gnome-doc-utils-0.20.10/xml2po/xml2po/xml2po.py.in source/gnome-doc-utils-0.20.10/xml2po/xml2po/xml2po.py.in
*** gnome-doc-utils-0.20.10/xml2po/xml2po/xml2po.py.in	2010-12-14 01:14:07.000000000 +0900
--- gnome-doc-utils-0.20.10/xml2po/xml2po/xml2po.py.in	2018-09-15 11:27:42.429101282 +0900
***************
*** 142,148 ****
          elif opt in ('-o', '--output'):
              output = arg
          elif opt in ('-v', '--version'):
!             print VERSION
              sys.exit(0)
          elif opt in ('-h', '--help'):
              usage(True)
--- 142,148 ----
          elif opt in ('-o', '--output'):
              output = arg
          elif opt in ('-v', '--version'):
!             print (VERSION)
              sys.exit(0)
          elif opt in ('-h', '--help'):
              usage(True)
