*** minidlna-1.2.1/minidlna.conf	2017-08-25 02:28:25.000000000 +0900
--- build/minidlna.conf	2017-12-09 02:29:53.385936238 +0900
***************
*** 15,21 ****
  #   + "V" for video  (eg. media_dir=V,/home/jmaggard/Videos)
  #   + "P" for images (eg. media_dir=P,/home/jmaggard/Pictures)
  #   + "PV" for pictures and video (eg. media_dir=PV,/home/jmaggard/digital_camera)
! media_dir=/opt
  
  # set this to merge all media_dir base contents into the root container
  # note: the default is no
--- 15,21 ----
  #   + "V" for video  (eg. media_dir=V,/home/jmaggard/Videos)
  #   + "P" for images (eg. media_dir=P,/home/jmaggard/Pictures)
  #   + "PV" for pictures and video (eg. media_dir=PV,/home/jmaggard/digital_camera)
! media_dir=/opt/minidlna
  
  # set this to merge all media_dir base contents into the root container
  # note: the default is no
