diff -uN comm-1.9.2/thunderbird.desktop thunderbird.new/thunderbird.desktop
--- comm-1.9.2/thunderbird.desktop	1970-01-01 09:00:00.000000000 +0900
+++ thunderbird.new/thunderbird.desktop	2011-01-25 12:35:02.000000000 +0900
@@ -0,0 +1,14 @@
+[Desktop Entry]
+Encoding=UTF-8
+Name=Mozilla Thunderbird Mail/News
+Comment=Read/Write Mail/News with Mozilla Thunderbird
+GenericName=Mail Client
+Exec=thunderbird %u
+Terminal=false
+X-MultipleArgs=false
+Type=Application
+Categories=Application;Network;Email;
+MimeType=text/html;text/xml;application/xhtml+xml;application/xml;application/vnd.mozilla.xul+xml;application/rss+xml;application/rdf+xml;
+StartupWMClass=Thunderbird-bin
+StartupNotify=true
+Comment[ja]=Mozilla Thunderbird メール/ニュース
