diff --git a/src/prediction/zero_query_dict.h b/src/prediction/zero_query_dict.h
index c11aa24d..6e24b0bf 100644
--- a/src/prediction/zero_query_dict.h
+++ b/src/prediction/zero_query_dict.h
@@ -147,6 +147,17 @@ class ZeroQueryDict {
       return iter;
     }
 
+    iterator &operator--() {
+      ptr_ -= kTokenByteSize;
+      return *this;
+    }
+
+    iterator operator--(int) {
+      const iterator tmp(ptr_, string_array_);
+      ptr_ -= kTokenByteSize;
+      return tmp;
+    }
+
     iterator &operator-=(ptrdiff_t n) {
       ptr_ -= n * kTokenByteSize;
       return *this;
