*** iputils-s20161105/doc/Makefile	2016-11-06 02:17:02.000000000 +0900
--- build/doc/Makefile	2018-05-15 15:39:40.967940164 +0900
***************
*** 43,49 ****
  $(MANFILES): index.db
  	@set -e; \
  	$(SETUP_TMPDIR); \
! 	nsgmls ../$< | sgmlspl ../docbook2man-spec.pl ""; \
  	mv $@ ..; \
  	$(CLEAN_TMPDIR)
  
--- 43,49 ----
  $(MANFILES): index.db
  	@set -e; \
  	$(SETUP_TMPDIR); \
! 	nsgmls ../$< | sgmlspl.pl ../docbook2man-spec.pl ""; \
  	mv $@ ..; \
  	$(CLEAN_TMPDIR)
  
