*** gst-plugins-bad-0.10.23/ext/vp8/gstvp8utils.h	2011-12-12 03:49:11.000000000 +0900
--- build/ext/vp8/gstvp8utils.h	2017-01-14 10:03:46.110445387 +0900
***************
*** 27,33 ****
  
  /* Some compatibility defines for older libvpx versions */
  #ifndef VPX_IMG_FMT_I420
! #define VPX_IMG_FMT_I420 IMG_FMT_I420
  #endif
  
  #ifndef VPX_PLANE_Y
--- 27,33 ----
  
  /* Some compatibility defines for older libvpx versions */
  #ifndef VPX_IMG_FMT_I420
! #define VPX_IMG_FMT_I420 (VPX_IMG_FMT_PLANAR | 2)
  #endif
  
  #ifndef VPX_PLANE_Y
