--- libunwind-1.2.1/configure.ac	2017-06-16 06:31:19.000000000 +0900
+++ build32/configure.ac	2017-12-19 14:19:51.669932620 +0900
@@ -31,8 +31,6 @@
   *) DLLIB="";;
 esac
 
-CHECK_ATOMIC_OPS
-
 dnl Checks for header files.
 AC_HEADER_STDC
 AC_CHECK_HEADERS(asm/ptrace_offsets.h endian.h sys/endian.h execinfo.h \
@@ -340,20 +338,6 @@
 fi
 AC_MSG_RESULT([$have__builtin_unreachable])
 
-AC_MSG_CHECKING([for __sync atomics])
-AC_LINK_IFELSE(
-  [AC_LANG_PROGRAM([[]], [[
-    __sync_bool_compare_and_swap((int *)0, 0, 1);
-    __sync_fetch_and_add((int *)0, 1);
-    ]])],
-  [have_sync_atomics=yes],
-  [have_sync_atomics=no])
-if test x$have_sync_atomics = xyes; then
-  AC_DEFINE([HAVE_SYNC_ATOMICS], [1],
-            [Defined if __sync atomics are available])
-fi
-AC_MSG_RESULT([$have_sync_atomics])
-
 CCASFLAGS="${CCASFLAGS} ${CPPFLAGS}"
 
 arch="$target_arch"
